emoji_data_python/replacement.py

```diff
@@ -11,23 +12,29 @@ def replace_colons(text: str, strip: bool=False) -> str:
     >>> emoji_data_python.replace_colons('Hello world ! :wave::skin-tone-3: :earth_africa: :exclamation:')
     'Hello world ! 👋🏼 🌍 ❗'
     """
-    from emoji_data_python import emoji_short_names
+    from emoji_data_python import emoji_short_names, EmojiChar
 
     def emoji_repl(matchobj) -> str:
-        match = matchobj.group(0)
-        codes = match.split(':')
-        res = ''
-        for code in codes:
-            if len(code) > 0:
-                try:
-                    res += emoji_short_names.get(code.replace('-', '_')).char
-                except AttributeError:
-                    if not strip:
-                        res += f':{code}:'
-
-        return res
-
-    return re.sub(r'\:[a-zA-Z0-9-_+]+\:(\:skin-tone-[2-6]\:)?', emoji_repl, text)
+        emoji_match = matchobj.group(1)
+        base_emoji = emoji_short_names.get(emoji_match.strip(':').replace('-', '_'))
+
+        if matchobj.lastindex == 2:
+            skin_tone_match = matchobj.group(2)
+            skin_tone = cast(EmojiChar, emoji_short_names.get(skin_tone_match.strip(':')))
```

Comment:
If what you need is to help type hinting I suggest we do
```suggestion
            skin_tone: EmojiChar = emoji_short_names.get(skin_tone_match.strip(':'))
```

Reply:
Done. Curious, why do you prefer this way of doing it?

Reply:
The `cast()` function can look as if it does something at runtime, I feel it impairs readability, I like to keep anything around types (which are not evaluated at runtime) outside of the code and behind the standard colons.
But now I think of it I'm not sure this step is necessary, the type should be inferred from the type of `emoji_short_names` without the need to specify it there. I think we can remove it altogether

Reply:
In this case, `emoji_short_names.get()`'s return type is actually `Optional[EmojiChar]`, because there's no guarantee ahead of time that the `skin-tone` we're asking for in `emoji_short_names` actually exists (**_we_** know it does, because we've defined it via our regex, but there's no strong guarantee - for instance, in future the keys may chage).

Therefore, `cast`ing it here tells the type hinting "Yes, this `.get` may actually be `Optional[EmojiChar]`, but trust me you should treat it as an `EmojiChar`." Whereas using the colon form tells the type hinter "This variable should expect `EmojiChar` to be fed into it" - which is actually wrong here, since we're feeding it `Optional[EmojiChar]`. This is demonstrated via `mypy`:

![image](https://user-images.githubusercontent.com/7358629/76083190-21693c00-6001-11ea-9cc8-3b711c44062c.png)
Top `mypy` call is current HEAD, the revert takes it back to using `cast`.

Reply:
And for completeness' sake, here's with no type hinting at all:

![image](https://user-images.githubusercontent.com/7358629/76083444-bec47000-6001-11ea-9ed3-f9e76b3863af.png)

Reply:
You're completely right, this isn't the same and `cast()` is very well needed here. Thanks for your clear explanations and sorry for bugging you about this even though you clearly know better than me. I have learnt something today thanks to you :)
Again, great PR, thanks for the contribution !

Reply:
You're completely right, this isn't the same and `cast()` is very well needed here. Thanks for your clear explanations and sorry for bugging you about this even though you clearly know better than me. I have learnt something today thanks to you :)
Care to revert back to the `cast()` implementation ?
Again, great PR, thanks for the contribution !

Reply:
No wuckas mate - I've reverted it back to using `cast`. Lemme know if you need anything else before merge, otherwise I'll leave the merge/release to you.